### PR TITLE
Rational numbers module

### PR DESCRIPTION
**Pull request by tudny**
Added rational numbers module
Added tests
Added LaTeX trait

### CODE REVIEW

File: src/rationals.rs [hunk 1]
```diff
@@ -0,0 +1,245 @@
+use std::fmt::Formatter;
+use super::latex::LaTeXable;
+
+/// This struct represents the rational number
+/// `n/d` where `n` is the numerator and `d` is the denominator.
+/// The sign of the rational number is always stored in the numerator.
```
Comment (mgr0dzicki) [inline src/rationals.rs]: Probably it is also worth mentioning here that gcd(numerator, denominator)=1 (required by comparisons, right?)

File: src/rationals.rs [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    }
+}
+
+impl PartialEq for Rational {
+    fn eq(&self, other: &Self) -> bool {
+        self.numerator == other.numerator && self.denominator == other.denominator
+    }
+}
+
+impl Eq for Rational {}
```
Comment (mgr0dzicki) [inline src/rationals.rs]: It's enough to use #[derive(Debug, Eq, PartialEq)]

File: src/rationals.rs [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+    fn sub(self, other: Self) -> Self {
+        self + -other
+    }
+}
+
+impl std::ops::Mul for Rational {
+    type Output = Self;
+
+    fn mul(self, other: Self) -> Self {
```
Comment (mgr0dzicki) [inline src/rationals.rs]: I'd use checked_mul and return Option<Self> (and same for other ops). Overflows aren't even checked in release build, I think it'd better to somehow notify the user and stop the calculations.

File: src/rationals.rs [hunk 4]
```diff
@@ -0,0 +1,245 @@
+use std::fmt::Formatter;
+use super::latex::LaTeXable;
+
+/// This struct represents the rational number
+/// `n/d` where `n` is the numerator and `d` is the denominator.
+/// The sign of the rational number is always stored in the numerator.
+#[derive(Debug)]
```
Comment (mgr0dzicki) [inline src/rationals.rs]: What about Clone and Copy?

**Review by mgr0dzicki — COMMENTED** [review]
Piękny kodzik

File: README.md [hunk 5]
```diff
@@ -17,13 +17,13 @@ One of its most powerful features is the ability to generate `LaTeX` code for th
...
-7. Implement complex operations on matrices.
+- 1.1. Implement rational numbers representation.
+- 1.2. Implement matrix module for given any number representation.
+- 1.3. Implement basic operations on matrices.
+- 1.4. Implement `LaTeX` code generation.
+- 2.1. Implement GUI.
+- 2.2. Extend the number representation module to allow for more sophisticated numbers.
+- 2.3. Implement complex operations on matrices.
 
 ---
```
Comment (mgr0dzicki) [inline README.md:29]: What about adding num in libraries?

**Review by mgr0dzicki — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (tudny) [reply README.md:29]: who asked

Comment (mgr0dzicki) [reply README.md:29]: https://docs.rs/num/latest/num/

**Pull request merged**